src/wifi/wifid-supplicant.c

```diff
@@ -2401,6 +2402,49 @@ static void supplicant_run(struct supplicant *s, const char *binary)
 	execve(argv[0], argv, environ);
 }
 
+static int supplicant_find(char **binary)
```

Comment:
This will work, but better and more POSIX, just change execve with execpve on line 2401 of supplicant_run and read its output for a possible error.